crates/daphne-server/src/router/mod.rs

```diff
@@ -324,21 +329,110 @@ where
             (task_id, resource)
         };
 
-        Ok(DapRequestExtractor(DapRequest {
+        let request = DapRequest {
             version,
             task_id,
             resource,
             payload: payload.to_vec(),
             media_type,
-            sender_auth: Some(sender_auth),
             taskprov,
-        }))
+        };
+
+        Ok(UnauthenticatedDapRequestExtractor(request))
     }
 }
 
+/// An axum extractor capable of parsing a [`DapRequest`].
+///
+/// This extractor asserts that the request is authenticated.
+#[derive(Debug)]
+struct DapRequestExtractor(pub DapRequest);
+
+#[async_trait]
+impl<S, B> FromRequest<S, B> for DapRequestExtractor
+where
+    S: DaphneService + Send + Sync,
+    B: HttpBody + Send + 'static,
+    <B as HttpBody>::Data: Send,
+{
+    type Rejection = (StatusCode, String);
+
+    async fn from_request(req: Request<B>, state: &S) -> Result<Self, Self::Rejection> {
+        let bearer_token = extract_header_as_string(req.headers(), http_headers::DAP_AUTH_TOKEN)
+            .map(BearerToken::from);
+        let cf_tls_client_auth = extract_header_as_string(req.headers(), "X-Client-Cert-Verified");
+
+        let request = UnauthenticatedDapRequestExtractor::from_request(req, state)
+            .await?
+            .0;
+
+        let is_taskprov = dbg!(state
```

Comment:
Docs for `dbg!()` say this is for "quick and dirty" debugging, so maybe we should remove it.